src/components/earth/Warning.tsx

```diff
@@ -5,21 +5,40 @@ import WarningIcon from "../../assets/svgs/Warning.svg?react";
 import { rowFlex } from "../../styles/flexStyles";
 import theme from "../../styles/theme";
 import DetailAnimation from "../animation/DetailAnimation";
+import { Pin } from "../../types/pin";
+import { climateIcons } from "../../constants/climateIcons";
+import { useNavigate } from "react-router-dom";
 
-const Warning = () => {
-  const [hovered, setHovered] = useState(false);
+interface WarningProps {
+  pin: Pin;
+}
 
+const Warning = ({ pin }: WarningProps) => {
+  const navigate = useNavigate();
+  const [hovered, setHovered] = useState(false);
+  const ClimateIcon = climateIcons.find(
+    (item) => item.id === pin.climateProblem[0]
+  )?.icon;
+  const ClimateTitle = climateIcons.find(
+    (item) => item.id === pin.climateProblem[0]
+  )?.label;
```

Comment:
```suggestion
  const [searchParams] = useSearchParams();
  const filterParam = searchParams.get("filter");
  const navigate = useNavigate();
  const [hovered, setHovered] = useState(false);
  
  const climate = filterParam && pin.climateProblem.includes(filterParam as Climate) 
    ? filterParam 
    : pin.climateProblem[0];

  const ClimateIcon = climateIcons.find(
    (item) => climate === item.id)?.icon;

  const ClimateTitle = climateIcons.find(
    (item) => climate === item.id)?.label;
```
이렇게 하면 현재 필터링 된 기후들을 제대로 불러올 수 있을 것 같아요!

Reply:
와~ 굿굿굿